keras_hub/src/models/gpt_oss/gpt_oss_attention.py

```diff
@@ -0,0 +1,351 @@
+            name="value",
+        )
+        self.value_dense.build(inputs_shape)
+
+        self.dropout_layer = keras.layers.Dropout(
+            rate=self.dropout,
+            dtype=self.dtype_policy,
+        )
+
+        self.output_dense = keras.layers.EinsumDense(
+            equation="bquh,uhm->bqm",
+            output_shape=(None, self._hidden_dim),
+            bias_axes="m",
+            kernel_initializer=self._kernel_initializer,
+            bias_initializer="zeros",
+            dtype=self.dtype_policy,
+            name="attention_output",
+        )
+        self.output_dense.build(
+            (None, None, self.num_query_heads, self._head_dim)
+        )
+
+        self.rotary_embedding_layer = RotaryEmbedding(
+            max_wavelength=self.rope_max_wavelength,
+            scaling_factor=self.rope_scaling_factor,  # YaRN scaling factor
+            rope_type="yarn",
+            beta_fast=32.0,
+            beta_slow=1.0,
+            original_max_position_embeddings=4096,
+            dtype=self.dtype_policy,
+        )
+
+        self.sinks = self.add_weight(
+            shape=(self.num_query_heads,),
+            initializer="random_normal",
+            dtype=self.dtype,
+            name="sinks",
+        )
+
+        self._dot_product_equation = "bquh,bkuh->buqk"
+        self._combine_equation = "buqk,bkuh->bquh"
+
+        self.built = True
+
+    def call(
+        self,
+        hidden_states,
+        attention_mask=None,
+        cache=None,
+        cache_update_index=None,
+        training=None,
+    ):
+        start_index = (
+            cache_update_index if cache_update_index is not None else 0
+        )
+
+        query = self.query_dense(hidden_states)
+
+        # Compute RoPE for queries (only
+        #  to first _rotary_dim dimensions)
+        if self._rotary_dim < self._head_dim:
+            query_rot = query[..., : self._rotary_dim]
+            query_rot = self.rotary_embedding_layer(
+                query_rot, start_index=start_index
+            )
+            query = ops.concatenate(
+                [query_rot, query[..., self._rotary_dim :]], axis=-1
+            )
+        else:
+            query = self.rotary_embedding_layer(query, start_index=start_index)
+
+        def _compute_key_value(x):
+            key, value = self.key_dense(x), self.value_dense(x)
+            # Compute RoPE for keys (only apply to first _rotary_dim dimensions)
+            if self._rotary_dim < self._head_dim:
+                key_rot = key[..., : self._rotary_dim]
+                key_rot = self.rotary_embedding_layer(
+                    key_rot, start_index=start_index
+                )
+                key = ops.concatenate(
+                    [key_rot, key[..., self._rotary_dim :]], axis=-1
+                )
+            else:
+                key = self.rotary_embedding_layer(key, start_index=start_index)
+            return key, value
+
+        if cache is not None:
```

Comment:
cache logic for KerasHub is located in causal_lm file.
example: https://github.com/keras-team/keras-hub/blob/c7bd26ddbbaf79e710ad35951fa62ccf1255784d/keras_hub/src/models/gemma/gemma_causal_lm.py#L217